specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx

```diff
@@ -99,6 +100,20 @@ export function QueryResults(props: QueryResultsProps): JSX.Element {
     canFetchMore,
   } = useFetchQueryResults(props);
 
+  const isEmbeddedCollectingEvent = schema.embeddedCollectingEvent;
+  const isEmbeddedPaleoContext = schema.embeddedPaleoContext;
+  const canMerge =
+    hasPermission('/record/merge', 'update') &&
+    hasTablePermission(table.name, 'update');
+  const canMergePaleoContext =
+    table.name === 'PaleoContext' && !isEmbeddedPaleoContext && canMerge;
+  const canMergeCollectingEvent =
+    table.name === 'CollectingEvent' && !isEmbeddedCollectingEvent && canMerge;
+  const canMergeOtherTables =
+    table.name !== 'PaleoContext' &&
+    table.name !== 'CollectingEvent' &&
+    canMerge;
+
```

Comment:
Good work, getting there. This can be generalized. More specifically, doesn't make sense to merge one-to-one records. via the schema config, this happens to be a one-to-one to collection object. Instead of these conditionals, add a check if the table being merged happens to be in a one-to-one relationship with any other table. If so, don't merge.